### PR TITLE
Add the VSCode client for `klsp`

### PR DESCRIPTION
**Pull request by radumereuta**
For testing the new `klsp` here https://github.com/runtimeverification/k/pull/3104.

Do not merge yet. It needs a bit more testing and cleaning.

This PR makes the connection to the klsp in the VSCode K extension.
To test it, just import the k-vscode project and run it in VSCode. If you have K in your path, everything should work automatically.
Requires the branch mentioned above.

### CODE REVIEW

**Comment by Baltoli** [comment]
Do you need a review on this @radumereuta, or does it still need cleaned up?

File: k-vscode/src/test/suite/extension.test.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// import * as myExtension from '../../extension';
+
+suite('Extension Test Suite', () => {
+	vscode.window.showInformationMessage('Start all tests.');
+
+	test('Sample test', () => {
+		assert.strictEqual(-1, [1, 2, 3].indexOf(5));
+		assert.strictEqual(-1, [1, 2, 3].indexOf(0));
+	});
+});
```
Comment (Baltoli) [inline k-vscode/src/test/suite/extension.test.ts:15]: What's going on here? These tests are placeholders and I don't think anything's getting imported

**Review by Baltoli — APPROVED** [review]
One minor comment in what looks like generated code, but otherwise LGTM

File: k-vscode/src/test/suite/extension.test.ts (hunk 1 above)
Comment (radumereuta) [reply k-vscode/src/test/suite/extension.test.ts:15]: Most of the code here is generated.
I only had to customize the part where we call the `klsp` executable.
In this case, we would write unit tests. But for now, we don't have that much functionality.
Thanks for the review.

**Pull request merged**